Report.md

```diff
@@ -374,3 +374,195 @@ steeply increasing runtime as the problem size and processor count increase.
+│     ├─ 0.011 MPI_Gather
+│     └─ 0.016 MPI_Scatter
+├─ 0.910 comp
+│  └─ 0.909 comp_large
+├─ 0.201 data_init_runtime
+└─ 0.440 correctness_check
+```
+
+#### Radix Sort Calltree:
+```
+# MPI Radix Sort
+143.856 main
+├─ 0.001 MPI_Comm_dup
+├─ 0.000 MPI_Finalize
+├─ 0.000 MPI_Finalized
+├─ 0.000 MPI_Init
+├─ 0.000 MPI_Initialized
+├─ 24.575 comm
+│  └─ 24.574 comm_small
+│     ├─ 0.000 MPI_Allgather
+│     ├─ 0.003 MPI_Allreduce
+│     └─ 24.570 MPI_Barrier
+├─ 118.738 comp
+│  ├─ 118.733 comp_large
+│  │  └─ 114.279 comm
+│  │     └─ 104.602 comm_large
+│  │        ├─ 5.384 MPI_Isend
+│  │        └─ 84.708 MPI_Recv
+│  └─ 0.004 comp_small
+├─ 0.000 correctness_check
+│  ├─ 0.000 MPI_Recv
+│  └─ 0.000 MPI_Send
+└─ 0.003 data_init_runtime 
+```
+### 3b. Collect Metadata
+
+Have the following code in your programs to collect metadata:
+```
+adiak::init(NULL);
+adiak::launchdate();    // launch date of the job
+adiak::libraries();     // Libraries used
+adiak::cmdline();       // Command line used to launch the job
+adiak::clustername();   // Name of the cluster
+adiak::value("algorithm", algorithm); // The name of the algorithm you are using (e.g., "merge", "bitonic")
+adiak::value("programming_model", programming_model); // e.g. "mpi"
+adiak::value("data_type", data_type); // The datatype of input elements (e.g., double, int, float)
+adiak::value("size_of_data_type", size_of_data_type); // sizeof(datatype) of input elements in bytes (e.g., 1, 2, 4)
+adiak::value("input_size", input_size); // The number of elements in input dataset (1000)
+adiak::value("input_type", input_type); // For sorting, this would be choices: ("Sorted", "ReverseSorted", "Random", "1_perc_perturbed")
+adiak::value("num_procs", num_procs); // The number of processors (MPI ranks)
+adiak::value("scalability", scalability); // The scalability of your algorithm. choices: ("strong", "weak")
+adiak::value("group_num", group_number); // The number of your group (integer, e.g., 1, 10)
+adiak::value("implementation_source", implementation_source); // Where you got the source code of your algorithm. choices: ("online", "ai", "handwritten").
+```
+
+They will show up in the `Thicket.metadata` if the caliper file is read into Thicket.
+#### Radix Sort Metadata:
+```
+# Metadata Columns
+Index(['cali.caliper.version', 'mpi.world.size', 'spot.metrics',
+       'spot.timeseries.metrics', 'spot.format.version', 'spot.options',
+       'spot.channels', 'cali.channel', 'spot:node.order', 'spot:output',
+       'spot:profile.mpi', 'spot:region.count', 'spot:time.exclusive',
+       'spot:time.variance', 'launchdate', 'libraries', 'cmdline', 'cluster',
+       'algorithm', 'programming_model', 'data_type', 'size_of_data_type',
+       'input_size', 'input_type', 'num_procs', 'scalability', 'group_num',
+       'implementation_source'],
+      dtype='object')
+```
+#### Metadata
+![alt text](image.png)
+![alt text](image-1.png)
+![alt text](image-2.png)
+![alt text](image-3.png)
```

Comment:
These links are broken and the images don't render. By the looks of your file structure, they need a `report_img/` prefix.

Reply:
oops i did edit that on my own branch but i guess i didnt push it. thanks for catching that!